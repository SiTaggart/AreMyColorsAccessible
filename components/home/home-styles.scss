@import '../../styles/utils';

.home {
  position: relative;

  @include breakpoint($small-screen) {
    min-height: 85%;
  }
}
