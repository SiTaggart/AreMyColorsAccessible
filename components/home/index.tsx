import React, { Component, ReactElement } from 'react';
import { SiteData, ColorCombinationTypes } from '../../types';
import Container from '../layouts/container';
import Layout from '../layouts/layout';
import Results from '../results';
import ColorInputs from '../colorInputs';

export interface HomeProps {
  handleBackgroundColorInputChange: (value: string) => void;
  handleBackgroundColorSliderChange: (hex: string) => void;
  handleTextColorInputChange: (value: string) => void;
  handleTextColorSliderChange: (hex: string) => void;
  siteData: SiteData;
}

class Home extends Component<HomeProps, {}> {
  public render(): ReactElement {
    const siteData: SiteData = this.props.siteData;
    const colorInfo: Partial<ColorCombinationTypes> = siteData.colorCombos[0].combinations[0];

    return (
      <Container variant="home">
        <Layout variant="small">
          <Results
            accessibility={colorInfo.accessibility}
            contrast={colorInfo.contrast}
            isLight={siteData.isLight}
          />
        </Layout>
        <Layout variant="large">
          <ColorInputs
            background={siteData.background}
            colorCombos={siteData.colorCombos}
            handleBackgroundColorInputChange={this.props.handleBackgroundColorInputChange}
            handleBackgroundColorSliderChange={this.props.handleBackgroundColorSliderChange}
            handleTextColorInputChange={this.props.handleTextColorInputChange}
            handleTextColorSliderChange={this.props.handleTextColorSliderChange}
            isLight={siteData.isLight}
            textColor={siteData.textColor}
          />
        </Layout>
      </Container>
    );
  }
}

export default Home;
