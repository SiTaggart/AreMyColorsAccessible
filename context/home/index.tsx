import * as React from 'react';
import Color from 'color';
import isEmpty from 'lodash/isEmpty';
import debounce from 'lodash/debounce';
import qs from 'query-string';
import ColorCombos, { ColorCombosTypes, ColorCombinationTypes } from 'color-combos';
import { SiteData } from '../../types';

export interface HomeContextInterface {
  siteData: SiteData;
  handleBackgroundColorInputChange: (value: string) => void;
  handleTextColorInputChange: (value: string) => void;
}

interface SiteDataProviderProps {
  children?: React.ReactElement;
  initialSiteData: SiteData | {};
}

const setInitialContext = (initialSiteData: SiteData | {}): SiteData => {
  let textColor = '#FFFFFF';
  let background = '#1276CE';
  let isLight = false;
  if (!isEmpty(initialSiteData) && 'textColor' in initialSiteData) {
    textColor = initialSiteData.textColor;
    background = initialSiteData.background;
    isLight = JSON.parse((initialSiteData.isLight as unknown) as string);
  }

  const initialCombos = ColorCombos([textColor, background]) as ColorCombosTypes[];
  return {
    background: background,
    textColor: textColor,
    isLight,
    colorCombos: initialCombos,
  };
};

const checkBackgroundLightness = (hex: string): boolean => {
  let light;
  try {
    light = Color(hex).isLight();
  } catch (e) {
    light = true;
  }
  return light;
};

const isValidColor = (value: string): Color | false => {
  let color: Color | false = false;
  try {
    color = Color(value);
  } catch (error) {
    console.error('ColorInput invalid color');
  }
  return color;
};

const createFakeCombination = (color: number[], hex: string): ColorCombinationTypes => {
  return {
    model: 'rgb',
    color: color,
    valpha: 1,
    hex: hex,
    contrast: 1,
    accessibility: { aa: false, aaLarge: false, aaa: false, aaaLarge: false },
  };
};

const createDuplicateCombination = (combos: ColorCombosTypes[]): ColorCombosTypes[] => {
  const color = combos[0].color != null ? combos[0].color : [];
  const dupeCombo = {
    ...combos[0],
    combinations: [createFakeCombination(color, combos[0].hex)],
  };
  return [dupeCombo, dupeCombo];
};

const HomeContext = React.createContext<HomeContextInterface | null>(null);

const useSiteData = (): HomeContextInterface => {
  const context = React.useContext(HomeContext);
  if (!context) {
    throw new Error('useSiteData must be used with SiteDataProvider');
  }
  return context;
};

const SiteDataProvider: React.FunctionComponent<SiteDataProviderProps> = (
  props: SiteDataProviderProps
): React.ReactElement => {
  const [siteData, setSiteData] = React.useState<SiteData>(
    setInitialContext(props.initialSiteData)
  );

  const [isInitial, setIsInitial] = React.useState<boolean>(false);

  const [state] = React.useMemo(
    (): [SiteData, React.Dispatch<SiteData>] => [siteData, setSiteData],
    [siteData]
  );

  const updateHash = debounce((): void => {
    const query = '?' + qs.stringify(state as {});
    window.history.pushState(state, 'Are My Colors Accessible', query);
  }, 200);

  React.useEffect((): void => {
    if (isInitial) {
      updateHash();
    } else {
      setIsInitial(true);
    }
  }, [state]);

  const setNewColorCombo = (textColor: string, backgroundColor: string): void => {
    let newCombos: ColorCombosTypes[] | false = ColorCombos([textColor, backgroundColor]);
    if (newCombos) {
      if (textColor === backgroundColor) {
        newCombos = createDuplicateCombination(newCombos);
      }
      setSiteData({
        ...state,
        background: backgroundColor,
        colorCombos: newCombos,
        isLight: checkBackgroundLightness(backgroundColor),
        textColor: textColor,
      });
    }
  };

  const handleBackgroundColorInputChange = (value: string): void => {
    setSiteData({
      ...state,
      background: value,
    });
    if (isValidColor(value)) {
      setNewColorCombo(state.textColor, value);
    }
  };

  const handleTextColorInputChange = (value: string): void => {
    setSiteData({
      ...state,
      textColor: value,
    });
    if (isValidColor(value)) {
      setNewColorCombo(value, state.background);
    }
  };

  return (
    <HomeContext.Provider
      value={{
        siteData: state,
        handleBackgroundColorInputChange: handleBackgroundColorInputChange,
        handleTextColorInputChange: handleTextColorInputChange,
      }}
      {...props}
    />
  );
};

export { HomeContext, useSiteData, SiteDataProvider };
