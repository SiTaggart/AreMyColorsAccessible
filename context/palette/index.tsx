import * as React from 'react';
import Color from 'color';
import qs from 'query-string';
import isEmpty from 'lodash/isEmpty';
import debounce from 'lodash/debounce';
import ColorCombos, { ColorCombosTypes } from 'color-combos';
import { PalettePageQueryString } from '../../pages/palette';

export interface PaletteContextProps {
  paletteData: PaletteState;
  setPaletteData: React.Dispatch<React.SetStateAction<PaletteState>>;
  handleColorChange: (value: string, index: number) => void;
  handleNewColor: (colors: string) => void;
}
interface PaletteState {
  colors: string[];
  colorCombos: ColorCombosTypes[];
  hasError: boolean;
}
interface PaletteDataProviderProps {
  children?: React.ReactElement;
  queryString?: PalettePageQueryString;
}

const convertColorStringsToColors = (colorStrings: string[]): Color[] | false => {
  let isValidColor = true;
  const colorTypes: Color[] = [];

  colorStrings.forEach((color: string): void => {
    try {
      colorTypes.push(Color(color));
    } catch (error) {
      isValidColor = false;
    }
  });

  if (isValidColor) {
    return colorTypes;
  } else {
    return isValidColor;
  }
};

const convertColorValuesToArray = (colors: string): string[] => {
  const colorsArr: string[] = colors.split(/[ ,]+/).filter(Boolean);
  const dedupedColors = colorsArr.filter(
    (color, index, self): boolean => self.indexOf(color) === index
  );
  return dedupedColors;
};

const isValidColor = (hex: string): Color | false => {
  let color: Color | false = false;
  try {
    color = Color(hex);
  } catch (error) {}
  return color;
};

const getColorCombos = (colors: string[]): ColorCombosTypes[] | false => {
  return ColorCombos(colors);
};

const getInitialState = (querystring: PalettePageQueryString | undefined): PaletteState => {
  let colors: string[] = [];
  let colorCombos: ColorCombosTypes[] = [];

  if (querystring !== undefined && !isEmpty(querystring)) {
    colors = querystring.colors;
    colorCombos = getColorCombos(colors) as ColorCombosTypes[];
  }

  return {
    colors: colors,
    colorCombos: colorCombos,
    hasError: false,
  };
};

const updateHash = debounce((state): void => {
  const query = '?' + qs.stringify({ colors: state.colors });
  window.history.pushState(state, 'Palette checker - Are My Colours Accessible', query);
}, 200);

const PaletteContext = React.createContext<PaletteContextProps | null>(null);

const usePaletteData = (): PaletteContextProps => {
  const context = React.useContext(PaletteContext);
  if (!context) {
    throw new Error('usePaletteData must be used with PaletteDataProvider');
  }
  return context;
};

const PaletteDataProvider: React.FC<PaletteDataProviderProps> = (
  props: PaletteDataProviderProps
): React.ReactElement => {
  const [paletteData, setPaletteData] = React.useState<PaletteState>(
    getInitialState(props.queryString)
  );

  const [isInitial, setIsInitial] = React.useState<boolean>(false);

  const [state] = React.useMemo(
    (): [PaletteState, React.Dispatch<PaletteState>] => [paletteData, setPaletteData],
    [paletteData]
  );

  React.useEffect((): void => {
    if (isInitial) {
      updateHash(state);
    } else {
      setIsInitial(true);
    }
  }, [state]);

  const mergeColorsWithState = (colors: string[]): string[] => {
    const filteredColors: string[] = colors.filter(
      (color): boolean => (state.colors as string[]).indexOf(color) < 0
    );
    return [...state.colors, ...filteredColors];
  };

  const updateColors = (colors: string[], isValidColor: boolean): void => {
    let newColorCombos: ColorCombosTypes[];
    if (isValidColor) {
      const combos = getColorCombos(colors);
      if (combos !== false) {
        newColorCombos = combos;
      } else {
        newColorCombos = state.colorCombos;
      }
    } else {
      newColorCombos = state.colorCombos;
    }
    setPaletteData({
      colors,
      colorCombos: newColorCombos,
      hasError: false,
    });
  };

  const handleColorChange = (value: string, index: number): void => {
    const newColors: string[] = [...state.colors];
    newColors[index] = value;
    updateColors(newColors, !!isValidColor(value));
  };

  const handleNewColor = (colors: string): void => {
    const colorsArray: string[] = convertColorValuesToArray(colors);
    const convertedColors: Color[] | false = convertColorStringsToColors(colorsArray);
    const mergedColors: string[] = mergeColorsWithState(colorsArray);

    if (convertedColors !== false) {
      updateColors(mergedColors, true);
    } else {
      setPaletteData({ ...state, hasError: true });
    }
  };

  return (
    <PaletteContext.Provider
      value={{
        paletteData: state,
        setPaletteData,
        handleColorChange,
        handleNewColor,
      }}
      {...props}
    >
      {props.children}
    </PaletteContext.Provider>
  );
};

export { PaletteContext, usePaletteData, PaletteDataProvider };
