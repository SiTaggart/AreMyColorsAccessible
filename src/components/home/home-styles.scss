.home {
    height: 85%;
    position: relative;

    .form {
        bottom: 0;
        left: 0;
        position: absolute;
        right: 0;
        width: 100%;
    }
}
