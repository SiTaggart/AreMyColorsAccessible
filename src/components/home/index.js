import React, { Component, PropTypes } from 'react';
import Colorable from 'colorable';
import './home-styles.scss';
import { Container, LayoutSmall, LayoutLarge } from '../layouts';
import Results from '../results';
import ColorInputs from '../colorInputs';

class Home extends Component {

    constructor(props) {
        super(props);
        this.setBackgroundColor = this.setBackgroundColor.bind(this);
        this.setForegroundColor = this.setForegroundColor.bind(this);
    }

    getColorInfo(background, foreground) {
        let colorInfo;
        try {
            colorInfo = Colorable([foreground, background])[0].combinations[0];
        } catch(e) {
            colorInfo = {
                contrast: 0,
                accessibility: {
                    aaa: false,
                    aa: false,
                    aaaLarge: false,
                    aaLarge: false
                }
            };
        }
        return colorInfo;
    }

    setBackgroundColor(hex) {
        this.props.setBackgroundColor(hex);
    }

    setForegroundColor(hex) {
        this.props.setForegroundColor(hex);
    }

    render() {
        const foreground = this.props.siteData.foreground;
        const background = this.props.siteData.background;
        const colorInfo = this.getColorInfo(background, foreground);

        return (
            <Container className="home">
                <LayoutSmall>
                    <Results {...colorInfo} />
                </LayoutSmall>
                <LayoutLarge>
                    <ColorInputs
                        background={this.props.siteData.background}
                        foreground={this.props.siteData.foreground}
                        isLight={this.props.siteData.isLight}
                        setForegroundColor={this.setForegroundColor}
                        setBackgroundColor={this.setBackgroundColor}
                    />
                </LayoutLarge>
                {this.props.children}
            </Container>
        );
    }
}

Home.propTypes = {
    children: PropTypes.node,
    setBackgroundColor: PropTypes.func,
    setForegroundColor: PropTypes.func,
    siteData: PropTypes.shape({
        background: PropTypes.string,
        foreground: PropTypes.string,
        isLight: PropTypes.bool
    })
};

export default Home;
