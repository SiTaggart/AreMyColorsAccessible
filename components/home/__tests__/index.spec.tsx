/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Home from '..';
import { SiteDataProvider } from '../../../context/home';

describe('Home', (): void => {
  it('renders without crashing', (): void => {
    ReactDOM.render(
      <SiteDataProvider
        initialSiteData={{
          background: '#000',
          textColor: '#fff',
          isLight: false,
          colorCombos: [],
        }}
      >
        <Home />
      </SiteDataProvider>,
      document.createElement('div')
    );
    const homeCmp = renderer
      .create(
        <SiteDataProvider
          initialSiteData={{
            background: '#000',
            textColor: '#fff',
            isLight: false,
            colorCombos: [],
          }}
        >
          <Home />
        </SiteDataProvider>
      )
      .toJSON();
    expect(homeCmp).toMatchSnapshot();
  });
});
