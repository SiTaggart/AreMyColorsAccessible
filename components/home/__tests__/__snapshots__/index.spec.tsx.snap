// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
.emotion-110 {
  position: relative;
}

@media (min-width:768px) {
  .emotion-110 {
    min-height: 85%;
  }
}

.emotion-28 {
  margin: 0 auto;
  max-width: 960px;
  padding: 0 1rem;
}

.emotion-26 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 2rem;
  text-align: center;
}

@media (min-width:768px) {
  .emotion-26 {
    margin-bottom: 3rem;
  }
}

@media (min-width:1024px) {
  .emotion-26 {
    margin-bottom: 5rem;
  }
}

.emotion-0 {
  font-weight: 700;
  font-size: 20vw;
  margin: 3rem 0;
  width: 100%;
}

@media (min-width:768px) {
  .emotion-0 {
    font-size: 18vw;
    margin: 4rem 0;
  }
}

@media (min-width:1024px) {
  .emotion-0 {
    font-size: 15vw;
    margin: 5rem 0;
  }
}

@media (min-width:1480px) {
  .emotion-0 {
    font-size: 9vw;
  }
}

.emotion-6 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-bottom: 2rem;
  width: 50%;
}

@media (min-width:768px) {
  .emotion-6 {
    width: 25%;
  }
}

.emotion-2 {
  font-weight: 700;
  font-size: 10vw;
  margin: 0;
}

@media (min-width:768px) {
  .emotion-2 {
    font-size: 6vw;
  }
}

@media (min-width:1024px) {
  .emotion-2 {
    font-size: 4vw;
  }
}

@media (min-width:1480px) {
  .emotion-2 {
    font-size: 3vw;
  }
}

.emotion-4 {
  font-size: 14pt;
  margin: 0;
}

.emotion-16 {
  font-size: 18pt;
  margin: 0;
}

.emotion-108 {
  margin: 0 auto;
  max-width: 1200px;
}

.emotion-106 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1rem;
  -webkit-transition: color 400ms ease-out;
  transition: color 400ms ease-out;
}

.emotion-66 {
  padding: 1rem;
}

@media (min-width:768px) {
  .emotion-66 {
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    width: 50%;
  }
}

.emotion-30 {
  display: inline-block;
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-32 {
  background: rgba(255,255,255,0.1);
  border: solid 1px;
  border-radius: 3px;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 2rem;
  width: 100%;
}

.emotion-64 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (min-width:1024px) {
  .emotion-64 {
    margin: 0 -0.5rem;
  }
}

.emotion-42 {
  width: 100%;
}

@media (min-width:1024px) {
  .emotion-42 {
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
    padding: 0 0.5rem;
    width: 33.3333%;
  }
}

.emotion-38 {
  -webkit-appearance: none;
  background: transparent;
  border-radius: 2px;
  color: currentColor;
  display: block;
  margin: 10px 0;
  width: 100%;
}

.emotion-38::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-38::-webkit-slider-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}

.emotion-38:focus::-webkit-slider-runnable-track {
  background: currentColor;
}

.emotion-38::-moz-range-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-38::-moz-range-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-38::-ms-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: transparent;
  border-color: transparent;
  border-width: 8px 0;
  color: transparent;
}

.emotion-38::-ms-fill-lower {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-38::-ms-fill-upper {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-38::-ms-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-38:focus::-ms-fill-lower {
  background: currentColor;
}

.emotion-38:focus::-ms-fill-upper {
  background: currentColor;
}

<main
  className="emotion-110 emotion-111"
>
  <div
    className="emotion-28 emotion-29"
  >
    <div
      className="emotion-26 emotion-27"
    >
      <h1
        aria-atomic="true"
        aria-live="polite"
        className="emotion-0 emotion-1"
        data-test="contrastResults-heading"
      >
        Yup
      </h1>
      <div
        className="emotion-6 emotion-7"
      >
        <h2
          className="emotion-2 emotion-3"
          data-test="contrastResult-rating-small"
        >
          AAA
        </h2>
        <p
          className="emotion-4 emotion-5"
        >
          Small Text - 4.5 
          <br />
          &lt; 14pt or 18px
        </p>
      </div>
      <div
        className="emotion-6 emotion-7"
      >
        <h2
          className="emotion-2 emotion-3"
          data-test="contrastResult-rating-bold"
        >
          AAA
        </h2>
        <p
          className="emotion-4 emotion-5"
        >
          <strong>
            Bold Text - 3.0 
            <br />
            &gt; 14pt or 18px
          </strong>
        </p>
      </div>
      <div
        className="emotion-6 emotion-7"
      >
        <h2
          className="emotion-2 emotion-3"
          data-test="contrastResult-rating-large"
        >
          AAA
        </h2>
        <p
          className="emotion-16 emotion-5"
        >
          Large Text - 3.0 
          <br />
          &gt; 18pt or 24px
        </p>
      </div>
      <div
        className="emotion-6 emotion-7"
      >
        <h2
          className="emotion-2 emotion-3"
        >
          21
        </h2>
        <p
          className="emotion-4 emotion-5"
        >
          Ratio
        </p>
      </div>
    </div>
  </div>
  <div
    className="emotion-108 emotion-29"
  >
    <div
      className="emotion-106 emotion-107"
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      <div
        className="emotion-66 emotion-67"
      >
        <label
          className="emotion-30 emotion-31"
          htmlFor="textColor"
        >
          Text Color
        </label>
        <input
          autoComplete="off"
          className="emotion-32 emotion-33"
          id="textColor"
          onChange={[Function]}
          style={
            Object {
              "borderColor": "#fff",
              "color": "inherit",
            }
          }
          type="text"
          value="#fff"
        />
        <div
          className="emotion-64 emotion-65"
          data-test="form-hsl-sliders"
        >
          <div
            className="emotion-42 emotion-43"
          >
            <div
              className="emotion-36 emotion-37"
            >
              <label
                className="emotion-30 emotion-31"
                htmlFor="textColor-hsl-Hue"
              >
                Hue 0Â°
              </label>
            </div>
            <div
              className="emotion-40 emotion-41"
            >
              <input
                className="emotion-38 emotion-39"
                id="textColor-hsl-Hue"
                max={360}
                min={0}
                onChange={[Function]}
                onInput={[Function]}
                type="range"
                value={0}
              />
            </div>
          </div>
          <div
            className="emotion-42 emotion-43"
          >
            <div
              className="emotion-36 emotion-37"
            >
              <label
                className="emotion-30 emotion-31"
                htmlFor="textColor-hsl-Saturation"
              >
                Saturation 0%
              </label>
            </div>
            <div
              className="emotion-40 emotion-41"
            >
              <input
                className="emotion-38 emotion-39"
                id="textColor-hsl-Saturation"
                max={100}
                min={0}
                onChange={[Function]}
                onInput={[Function]}
                type="range"
                value={0}
              />
            </div>
          </div>
          <div
            className="emotion-42 emotion-43"
          >
            <div
              className="emotion-36 emotion-37"
            >
              <label
                className="emotion-30 emotion-31"
                htmlFor="textColor-hsl-Lightness"
              >
                Lightness 100%
              </label>
            </div>
            <div
              className="emotion-40 emotion-41"
            >
              <input
                className="emotion-38 emotion-39"
                id="textColor-hsl-Lightness"
                max={100}
                min={0}
                onChange={[Function]}
                onInput={[Function]}
                type="range"
                value={100}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="emotion-66 emotion-67"
      >
        <label
          className="emotion-30 emotion-31"
          htmlFor="background"
        >
          Background
        </label>
        <input
          autoComplete="off"
          className="emotion-32 emotion-33"
          id="background"
          onChange={[Function]}
          style={
            Object {
              "borderColor": "#fff",
              "color": "inherit",
            }
          }
          type="text"
          value="#000"
        />
        <div
          className="emotion-64 emotion-65"
          data-test="form-hsl-sliders"
        >
          <div
            className="emotion-42 emotion-43"
          >
            <div
              className="emotion-36 emotion-37"
            >
              <label
                className="emotion-30 emotion-31"
                htmlFor="background-hsl-Hue"
              >
                Hue 0Â°
              </label>
            </div>
            <div
              className="emotion-40 emotion-41"
            >
              <input
                className="emotion-38 emotion-39"
                id="background-hsl-Hue"
                max={360}
                min={0}
                onChange={[Function]}
                onInput={[Function]}
                type="range"
                value={0}
              />
            </div>
          </div>
          <div
            className="emotion-42 emotion-43"
          >
            <div
              className="emotion-36 emotion-37"
            >
              <label
                className="emotion-30 emotion-31"
                htmlFor="background-hsl-Saturation"
              >
                Saturation 0%
              </label>
            </div>
            <div
              className="emotion-40 emotion-41"
            >
              <input
                className="emotion-38 emotion-39"
                id="background-hsl-Saturation"
                max={100}
                min={0}
                onChange={[Function]}
                onInput={[Function]}
                type="range"
                value={0}
              />
            </div>
          </div>
          <div
            className="emotion-42 emotion-43"
          >
            <div
              className="emotion-36 emotion-37"
            >
              <label
                className="emotion-30 emotion-31"
                htmlFor="background-hsl-Lightness"
              >
                Lightness 0%
              </label>
            </div>
            <div
              className="emotion-40 emotion-41"
            >
              <input
                className="emotion-38 emotion-39"
                id="background-hsl-Lightness"
                max={100}
                min={0}
                onChange={[Function]}
                onInput={[Function]}
                type="range"
                value={0}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
